### ISSUE TITLE
Output "Channel creation failed"

### ISSUE THREAD

**Issue by SpotlightKid** [jack-mixer/jack_mixer#18]
Happens when adding an output channel with a name, which contains a character, which is not valid in a CSS class name, e.g. whitespace.

If you add this after line 524 to `jack_mixer.py`:

       import traceback; traceback.print_exc()

you'll get the following traceback when adding an output channel with the name "out 1"

```
Traceback (most recent call last):
  File "./jack_mixer.py", line 523, in add_output_channel
    self.add_output_channel_precreated(channel)
  File "./jack_mixer.py", line 555, in add_output_channel_precreated
    channel.realize()
  File "/home/chris/work/jack_mixer/channel.py", line 591, in realize
    set_background_color(self.label_name_event_box, self.channel.name + 'label',
  File "/home/chris/work/jack_mixer/channel.py", line 53, in set_background_color
    css_provider.load_from_data(css.encode('utf-8'))
gi.repository.GLib.GError: gtk-css-provider-error-quark: <data>:2:9Expected a valid selector (1)
```

A quick fix would be something like this:

```diff
diff --git a/channel.py b/channel.py
index b163edf..4698f77 100644
--- a/channel.py
+++ b/channel.py
@@ -43,11 +43,12 @@ screen = Gdk.Screen.get_default()
 context.add_provider_for_screen(screen, css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION)
 
 def set_background_color(widget, name, color_string):
+    css_name = name.strip().replace(' ', '_')
     css = """
     .%s {
         background-color: %s
     }
-""" % (name, color_string)
+""" % (css_name, color_string)
 
     css_provider = Gtk.CssProvider()
     css_provider.load_from_data(css.encode('utf-8'))
@@ -56,7 +57,7 @@ def set_background_color(widget, name, color_string):
     context.add_provider_for_screen(screen, css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION)
 
     widget_context = widget.get_style_context()
-    widget_context.add_class(name)
+    widget_context.add_class(css_name)
 
 def random_color():
     from random import uniform, seed
```

but of course the same error would happen with other invalid characters, like non-ASCII chars, punctuation etc. Maybe it would be better to not derive the CSS class name from the channel name but from a fixed string with a number suffix (e.g. "output-1")?

**Comment by dsheeler** [comment]
I thought I already implemented the quick fix for spaces in the name, so surprised to see this. I’ll check it out.

**Comment by SpotlightKid** [comment]
Yes, it's in `ControlGroup.update`, but here it is called from `OutputChannel.realize`.

**Comment by dsheeler** [comment]
I implemented your idea of using a fixed string with a number suffix d6af8a48f74ca161a49614e4948f2e9a6dd01bc4
